zenoh-flow/src/runtime/dataflow/instance/runners/connector.rs

```diff
@@ -53,7 +57,52 @@ impl ZenohSender {
         })
     }
 
-    pub async fn run(&self) -> ZFResult<()> {
+    pub async fn run_stoppable(&self, stop: Receiver<()>) -> ZFResult<()> {
+        loop {
+            let run = async {
+                match self.run().await {
+                    Ok(_) => RunAction::RestartRun(None),
+                    Err(e) => RunAction::RestartRun(Some(e)),
+                }
+            };
+            let stopper = async {
+                match stop.recv().await {
+                    Ok(_) => RunAction::Stop,
+                    Err(e) => RunAction::StopError(e),
+                }
+            };
+
+            match run.race(stopper).await {
+                RunAction::RestartRun(e) => {
+                    log::error!("The run loop exited with {:?}, restarting...", e);
```

Comment:
I think it would be nice to prefix the logging messages with the id of the connector (and the node, in general). It could ease debugging.

`log::error!("[Connector: {}] The run loop exited with {:?}, restarting…", self.id, e);`